pandas/core/groupby/groupby.py

```diff
@@ -901,8 +901,13 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
                 FutureWarning,
                 stacklevel=find_stack_level(),
             )
-        if isinstance(keys, list) and len(keys) == 1:
-            # GH#42795 - when keys is a list, return tuples even when length is 1
+        if (
+            isinstance(keys, list)
+            and len(keys) == 1
+            or isinstance(keys, ops.BaseGrouper)
+            and len(keys.names) == 1
```

Comment:
Does this result in tuples even in the case of e.g. `.groupby("a")[["a", "b", "c"]]`?

Reply:
My bad, you're right. I may need to figure out some other way.

Reply:
In pandas.core.groupby.generic.DataFrameGroupBy._gotitem, we pass `self.grouper` for the `keys` argument. I think if you instead pass `self.keys` you don't need to modify anything else.

However this leads to issues with our `nth` implementation that should be fixed by #53519. Once that PR is merged, we should be able to fix this.

Reply:
Thanks for saving me time! I confirm that this does fix the issue. Will revert and commit this suggestion.